cypress/defs.bzl

```diff
@@ -1 +1,96 @@
 "Public API re-exports"
+
+load("@aspect_rules_cypress//cypress/private:cypress_module_test.bzl", cypress_module_test_lib = "lib")
+load("@aspect_rules_cypress//cypress/private:cypress_cli_test.bzl", cypress_cli_test_lib = "lib")
+load("@aspect_rules_js//js:libs.bzl", "js_binary_lib")
+load("@aspect_bazel_lib//lib:directory_path.bzl", "directory_path")
+
+_cypress_module_test = rule(
+    doc = """Runs tests against the Cypress test runner.""",
+    attrs = cypress_module_test_lib.attrs,
+    implementation = cypress_module_test_lib.implementation,
+    test = True,
+    toolchains = js_binary_lib.toolchains + ["@aspect_rules_cypress//cypress:toolchain_type"],
+)
+
+_cypress_cli_test = rule(
+    doc = """Identical to js_test, except entry point is set to cypress CLI with the cypress toolchain available""",
+    implementation = cypress_cli_test_lib.implementation,
+    attrs = cypress_cli_test_lib.attrs,
+    test = True,
+    toolchains = js_binary_lib.toolchains + ["@aspect_rules_cypress//cypress:toolchain_type"],
+)
+
+def cypress_cli_test(name, cypress = "//:node_modules/cypress", **kwargs):
```

Comment:
I improved the vendoring of npm packages that rules_js rule sets depend on so only a lock file is vendored now for any given version of the tool. https://github.com/aspect-build/rules_jest/blob/main/scripts/mirror_release.sh.

wdyt of switching to that so you can run this rule without referencing a cypress the user links?

Also related is vendoring in the hashes for the version as well so the integrity hashes are automatic instead of manual.

Reply:
We vendor the integrity hashes in rules_esbuild as an example: https://github.com/aspect-build/rules_esbuild/blob/main/esbuild/private/versions.bzl

Reply:
I'm open to making that change. Would you be comfortable with leaving that for a follow up PR? Also open to doing this now, but I think I'd like to get this into people's hands sooner rather than later personally

Reply:
Yup. That one is fine as a follow-up. Will be easier to reason about the change that way as well.